vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1425,3 +1443,15 @@ def get_kv_cache_spec(self) -> KVCacheSpec:
                     f"Unknown attention type: {attn_module.attn_type}")
 
         return kv_cache_spec
+
+    def _disable_spec_decode(self, req_idx: int,
+                             sampling_metadata: SamplingMetadata) -> bool:
+        return ((sampling_metadata.top_p
+                 and sampling_metadata.top_p[req_idx] < 1.0)
+                or (sampling_metadata.top_k
+                    and sampling_metadata.top_k[req_idx] > 0)
+                or (sampling_metadata.min_p
+                    and sampling_metadata.min_p[req_idx] > 0.0)
+                or (sampling_metadata.frequency_penalties[req_idx] != 0.0)
+                or (sampling_metadata.presence_penalties[req_idx] != 0.0)
+                or (sampling_metadata.repetition_penalties[req_idx] != 1.0))
```

Comment:
Maybe we should also add logprobs, etc.?

Reply:
^ @LiuXiaoxuanPKU

Reply:
I disable spec decode if logprobs is needed as [here](https://github.com/vllm-project/vllm/blob/cd24f5ec1182853403bbc6b4c7dc85debb239ea1/vllm/v1/spec_decode/utils.py#L21).